dosbox-staging/dosbox-staging
# Country codes cleanup, fix incorrect country set in some cases

# Description

1. Cleaned up country codes
    - added all country codes missing from different DOSes, _MS-DOS 6.22_ having the highest priority in case of conflicts/mismatches; also checked code pages supported by _PC DOS 2000_, _DR DOS 7.03_,  _FreeDOS 1.3_, _PTS DOS Pro 2000_
    - documented the origin of various country codes, documented all the deviation
    - added some missing locale information

2. Fixed a problem with wrong country code being set in some cases
The config option description claims _If set to 0, the country code corresponding to the selected keyboard layout will be used_. This was not entirely true - in fact, it was trying to guess country number from keyboard layout if there was no default keyboard layout associated with the given country. This is now fixed.


## Related issues

https://github.com/dosbox-staging/dosbox-staging/issues/3128 (fixes the PR only partially!)


# Manual testing

You can set `country = code` in the config file (codes suggested to check in the table below). Once DOSBox is started, type command `dir` and check that time/date is printed out according to `COUNTRY_INFO` structure from `dos.cpp` file

| country name | code   |
| ------------- | ------------- |
| Canada_English | 4   |
| South_Africa   | 27  |
| Mexico         | 52  |
| Chile          | 56  |
| Colombia       | 57  |
| Venezuela      | 58  |
| New_Zealand    | 64  |
| Vietnam        | 84  |
| Niger          | 227 |
| Benin          | 229 |
| Ireland        | 353 |
| Malta          | 356 |
| Armenia        | 374 |
| Slovakia       | 421 |
| Ecuador        | 593 |
| Hong_Kong      | 852 |
| Mongolia       | 976 |
| Tadjikistan    | 992 |
| Uzbekistan     | 998 |


# Checklist

I have:
